lib/order.rb

```diff
@@ -0,0 +1,75 @@
+require_relative 'menu'
+
+class Order
+
+  attr_reader :ordered_dishes, :order_total
+  PART_I_LENGTH = 42
+  PART_II_LENGTH = 8
+
+  def initialize
+    @ordered_dishes = []
+    @order_total = 0
+  end
+
+  def add_to_order(dish,quantity)
+    @ordered_dishes << {dish:dish,quantity:quantity}
+  end
+
+  def display_order_on_email
+    display = display_order_title
+    display += display_order_body
+    display += display_order_total
+  end
+
+  def display_order_on_sms
+    display = display_order_on_email
+    totals_line = "-"*(PART_I_LENGTH + PART_II_LENGTH)
+    new_totals_line = "-"*((PART_I_LENGTH + PART_II_LENGTH)/2)
+    display.gsub!("\n\n","\n")
+    display.gsub!(" ","")
+    display.gsub!("£"," £ ")
+    display.gsub!(totals_line,new_totals_line)
+    display
+  end
+
+  private
+
+  def display_order_title
+    title = "Order Details"
+    title_spacing = (PART_I_LENGTH + PART_II_LENGTH - title.length)/2
+    " "*title_spacing + "#{title}\n\n"
+  end
+
+  def display_order_body
+    display = ''
+    ordered_dishes.each{|dish| display += add_line_to_order_display(dish)}
+    display
+  end
+
+  def display_order_total
```

Comment:
Metrics/AbcSize: Assignment Branch Condition size for display_order_total is too high. [25.08/15]